### ISSUE TITLE
Parsing bug in some expressions

### ISSUE THREAD

**Issue by AugLuk** [rhaiscript/rhai#713]
Example:
```rust
// Error
let v = [0.5];
(v[0].sin()).cos()
```
```
EXCEPTION: Function not found: sin (i64)
```
It also produces an analogous exception when using other float functions instead of *sin* and *cos*.
```rust
// Error
let v = [0.5];
(v[0].atan(1.0)).log(2.0)
```
```
EXCEPTION: Function not found: atan (i64, f64)
```
However, using a float instead of an array of floats, removing the unnecessary brackets, or removing the second float function doesn't result in the exception.
```rust
// Ok
let v = 0.5;
(v.sin()).cos()
```
```rust
// Ok
let v = [0.5];
v[0].sin().cos()
```
```rust
// Ok
let v = [0.5];
(v[0].sin())
```
Tested in:
* Rhai Playground
* With Rhai 1.13.0 crate in x86_64-unknown-linux-gnu debug and release builds

**Comment by schungx** [comment]
Hhhmmm... that seems to be a bug with parsing.

Removing the parentheses work correctly... so this ia a parsing problem.

**Comment by AugLuk** [comment]
Looks like extra brackets can also produce other types errors where they shouldn't.
On debug builds the maximum expression nesting depth in functions should be 16, but:
```rust
let engine = Engine::new();
let mut scope = Scope::new();
let script = r#"fn foo() {
    let v = 0.5;
    if -((-(((v)).tan())).sin()) != 93.29292920983892 {
        return 4;
    } else {
        return 1;
    }
}"#;
let ast = engine.compile(script).unwrap();
let bar = engine.call_fn::<i64>(&mut scope, &ast, "foo", ()).unwrap();
println!("{bar}");
```
```
ParseError(ExprTooDeep, 3:15)
```
Removing either minus sign fixes it tho:
```rust
    if -((-(((v)).tan())).sin()) != 93.29292920983892 {
       ^  ^
```

**Comment by schungx** [comment]
> Looks like extra brackets can also produce other types errors where they shouldn't.

Probably.  It is an error in the parser so the expression is constructed incorrectly.  It may affect other places.

But it is unlikely to affect max expression depth.  Remember, each level of parentheses is a level, and the unary minus operator is itself a level.  You may have just hit the limit.

**Comment by schungx** [comment]
The fix has landed in #714
